src/StreamrClient.js

```diff
@@ -441,18 +509,37 @@ export default class StreamrClient extends EventEmitter {
         }
 
         if (options.groupKeys) {
-            this.options.subscriberGroupKeys[options.stream] = options.groupKeys
+            Object.keys(options.groupKeys).forEach((publisherId) => {
+                EncryptionUtil.validateGroupKey(options.groupKeys[publisherId])
+                if (!this.options.subscriberGroupKeys[options.stream]) {
+                    this.options.subscriberGroupKeys[options.stream] = {}
+                }
+                this.options.subscriberGroupKeys[options.stream][publisherId] = {
+                    groupKey: options.groupKeys[publisherId],
+                    start: Date.now()
```

Comment:
it's possible these will get different values for `start`. Might not matter, but perhaps should do `const now = Date.now()` *outside* the `forEach` loop so they all get the same `start`?